src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp

```diff
@@ -101,6 +118,134 @@ class PolicyManager final : public Singleton<PolicyManager>
+            }
+        }
+    }
+
+    /**
+     * @brief Validates and configures the vulnerability detection based on the provided JSON object.
+     *
+     * This function takes a JSON object as input, which is expected to contain configuration
+     * information for an vulnerability detection. It validates the JSON object to ensure it contains the
+     * required fields and has valid values.
+     * If the validation passes, no exception is throw.
+     *
+     * @param vdObj A constant reference to a JSON object representing the vulnerability detection's configuration.
+     *
+     * @note This function assumes that the provided JSON object follows a specific format.
+     * @note If validation fails, this function throw exceptions.
+     *
+     */
+
+    void validateAndConfigureVulnerabilityDetection(const nlohmann::json& vdObj)
+    {
+        bool isValidUrl;
+
+        if (!vdObj["vulnerability-detection"].contains("enabled"))
+        {
+            throw std::runtime_error("Missing enabled field.");
+        }
+
+        if (!vdObj["vulnerability-detection"].contains("index-status"))
+        {
+            throw std::runtime_error("Missing index-status field.");
+        }
+
+        if (vdObj["vulnerability-detection"].contains("feed-update-interval"))
+        {
+            m_feedUpdateTime = Utils::parseStrToTime(vdObj["vulnerability-detection"]["feed-update-interval"]);
+        }
+
+        if (vdObj["vulnerability-detection"].contains("offline-url"))
+        {
+            isValidUrl = Utils::startsWith(vdObj["vulnerability-detection"]["offline-url"], "file") ||
+                         Utils::startsWith(vdObj["offline-url"], "https") ||
+                         Utils::startsWith(vdObj["offline-url"], "https");
+        }
+
+        m_vulDetectEnabled = m_vulDetectEnabled && Utils::parseStrToBool(vdObj["vulnerability-detection"]["enabled"]);
+
+        m_indexerEnabled = m_indexerEnabled && Utils::parseStrToBool(vdObj["vulnerability-detection"]["index-status"]);
+
+        if (isValidUrl)
+        {
+            m_urlFeed = std::move(vdObj["vulnerability-detection"].contains("offline-url"));
+        }
+    }
+
+    /**
+     * @brief Validates and configures the indexer based on the provided JSON object.
+     *
+     * This function takes a JSON object as input, which is expected to contain configuration
+     * information for an indexer. It validates the JSON object to ensure it contains the
+     * required fields and has valid values, and based of the previous configuration of the vulnerability detection.
+     * If the validation passes, no exception is throw.
+     *
+     * @param idObj A constant reference to a JSON object representing the indexer's configuration.
+     *
+     * @note This function assumes that the provided JSON object follows a specific format.
+     * @note If validation fails, this function throw exceptions.
+     *
+     */
+
```

Comment:
Typos:
- `based of`
- `is throw.`
- ` this function throw exceptions.`

You can use Grammarly for this.